compiler/dcalc/ast.ml

```diff
@@ -87,36 +92,41 @@ type binop =
   | Map
   | Concat
   | Filter
+[@@ deriving show]
 
 type log_entry = VarDef of typ | BeginCall | EndCall | PosRecordIfTrueBool
+[@@ deriving show]
 
 type unop =
   | Not
   | Minus of op_kind
-  | Log of log_entry * (Utils.Uid.MarkedString.info list[@opaque])
+  | Log of log_entry * (Utils.Uid.MarkedString.info list [@opaque])
   | Length
   | IntToRat
   | GetDay
   | GetMonth
   | GetYear
+[@@ deriving show]
```

Comment:
Why do you introduce the `show` derivation and not use `Dcalc.Print` ?

Reply:
fixed